### PR TITLE
Add torch and cuda args to dockerfile

### PR DESCRIPTION
**Pull request by mmeendez8**
Fixed default versions gives problems, acutal dockerfile crushes since it install a not compatible mmcv version:
`AssertionError: MMCV==1.1.5 is used but incompatible. Please install mmcv>=1.2.4, <=1.3.
`

### CODE REVIEW

**Comment by mmeendez8** [comment]
This might not completely fix the problem, since docker CUDA version depends on tensor rt base image (see https://docs.nvidia.com/deeplearning/tensorrt/container-release-notes/index.html).

**Comment by grimoire** [comment]
Thanks for the contribution!
I did want to update the base image to a new version. But some conversions failed when I am trying to update tensorrt:20.09-py3. I will try the latest one this weekend.

**Comment by mmeendez8** [comment]
This worked for me with CUDA 11.2, just tested and run the output .engine with latest deepstream version.
Thanks to you for this amazing work!

File: README.md [hunk 1]
```diff
@@ -45,10 +45,17 @@ python setup.py develop
...
-# cuda10.2 tensorrt7.0 pytorch1.6
+# cuda11.2 tensorrt7.0 pytorch1.6
 sudo docker build -t mmdet2trt_docker:v1.0 docker/
 ```
 
+You can also specify CUDA, Pytorch and Torchvision versions with docker build args by:
+
+```shell
+# cuda10.2 tensorrt7.0 pytorch1.6
+sudo docker build -t mmdet2trt_docker:v1.0 --build-arg CUDA=11.2 --build-arg TORCH_VERSION=1.6.0 --build-arg TORCHVISION_VERSION=0.7.0 --docker/
```
Comment (grimoire) [inline README.md:56]: It seems that there is no prebuilt mmcv-full for cuda 11.2.

**Comment by grimoire** [comment]
cub has been include in cuda11, which is conflict with the cub in amirstan_plugin. Two stage model conversion failed in  tensorrt:20.09-py3(cuda11). This PR will be merged after I fix this.

**Comment by devloper13** [comment]
So what is the fix? Except 11.2, does it owrk for any version (from 10.0 to 11.1)? I get the same error AssertionError: MMCV==1.1.5 is used but incompatible. Please install mmcv>=1.2.4, <=1.3.

**Comment by mmeendez8** [comment]
You can use this branch and set cuda and pytorch versions in docker build, it should work (it worked for me).

**Comment by devloper13** [comment]
> You can use this branch and set cuda and pytorch versions in docker build, it should work (it worked for me).

I followed your suggested changes yet I'm stick with some errors. First, while building docker, mmpycocotools failed to build and also mmlvis. But the build still completed and upon running the command: sudo docker run --gpus all -it --rm -v ${your_data_path}:${bind_path} mmdet2trt_docker:v1.0 ${bind_path}/config.py ${bind_path}/checkpoint.pth ${bind_path}/output.trt, it gave me an error: pkg_resources.DistributionNotFound: The 'mmdet2trt' distribution was not found and is required by the application

Any ideas?

**Comment by mmeendez8** [comment]
> > You can use this branch and set cuda and pytorch versions in docker build, it should work (it worked for me).
> 
> I followed your suggested changes yet I'm stuck with some errors. First, while building docker, mmpycocotools failed to build and also mmlvis. But the build still completed and upon running the command: sudo docker run --gpus all -it --rm -v ${your_data_path}:${bind_path} mmdet2trt_docker:v1.0 ${bind_path}/config.py ${bind_path}/checkpoint.pth ${bind_path}/output.trt, it gave me an error: pkg_resources.DistributionNotFound: The 'mmdet2trt' distribution was not found and is required by the application
> 
> Any ideas?

It seems there was some problem installing mmdet2trt... I cannot tell you much from that information.  Check docker build output after the step:
```
### install mmdetection-to-tensorrt
RUN cd /root/space/mmdetection-to-tensorrt &&\
    python3 setup.py develop
```

Or get into docker container and try to figure out if mmdet2trt was properly installed.

**Comment by mmeendez8** [comment]
Any news about this?

**Comment by grimoire** [comment]
Sorry for the late reply
As MMCV only provides packages with cuda<=11.1, could you downgrade the base image and update the default config?

**Pull request merged**